### ISSUE TITLE
feature request: allow changing ndarray WRITABLE flag after ndarray construction

### ISSUE THREAD

**Issue by HDembinski** [boostorg/python#171]
I want to write a function which creates, fills, and returns an ndarray in C++. The ndarray owns its memory, but should be non-writable for the user. I would therefore like to clear the WRITABLE flag after I finished filling it in my function. I studied the header and I see no way of doing this right now, because flags are not writable after creating an ndarray. For most flags this makes sense, the WRITABLE flag seems to be the only one which you may want to change manually.

Is it possible to add this feature? Best regards!

**Comment by HDembinski** [comment]
Any chance that this will be implemented? Would you review a PR if I prepare one?

**Comment by stefanseefeld** [comment]
Yes, it looks like a simple `ndarray::set_flags()` method is missing, right ?

**Comment by stefanseefeld** [comment]
I'd rather not hand-hold users of the API. So if this is possible in the NumPy C API, I'm inclined to make it accessible in Boost.Python directly. But of course, proper documentation (including warning signs) would be helpful.

**Comment by HDembinski** [comment]
Ok!

**Comment by HDembinski** [comment]
I submitted a PR. Some tests are failing, but this seems unrelated to the new code...
